packages/aws-cdk-lib/aws-lambda/lib/function.ts

```diff
@@ -385,6 +455,30 @@ export interface FunctionOptions extends EventInvokeConfigOptions {
    * @default Auto
    */
   readonly runtimeManagementMode?: RuntimeManagementMode;
+
+  /**
+   * Sets the log group name for the function.
+   * @default `/aws/lambda/${this.functionName}` default log group name created by Lambda
+   */
+  readonly logGroup?: logs.ILogGroup;
+
+  /**
+   * Sets the logFormat for the function.
+   * @default Text format
+   */
+  readonly logFormat?: string;
```

Comment:
Why is this not typed (enum)? Same for log level.

Reply:
haa i see.

Reply:
A breaking change can be avoided by setting the type to `LogFormat | keyof typeof LogFormat` which would allow using either the enum or type-safe strings as the value.

Suggested change in #28127 which aims to solve this.

Reply:
Thank you for your suggestion.
if LogFormat type is changed to `logFormat?: LogFormat | keyof typeof LogFormat;`, An error like the following occurs in the code: `logFormat: 'Text'`.
`Type '"Text"' is not assignable to type 'LogFormat | "TEXT" | "JSON" | undefined'. Did you mean '"TEXT"'?ts(2820)`
Would you mind telling me a solution that accepts both `lambda.LogFormat.TEXT` and string `'Text'`.

Reply:
I might solve this problem by myself. I suggest changes in #28127.